### PR TITLE
string improvement: remove double and on the end spaces

### PR DESCRIPTION
**Pull request by davidsonsns**
### Background
Sometimes the `fake.string` was generating random strings with double spaces or spaces at the end of the content.

It was generating many React Native errors when it was happening because of the `react-testing-library`(https://www.native-testing-library.com/), library that we're using to make our tests, don't make this validation without send normalization text options(https://www.native-testing-library.com/docs/api-queries#normalization).

Talking with @ndhoule about it, he told that it could be a not intentional behavior by this function.

The proposed here is to avoid strings with double spaces or spaces at the end.

### CODE REVIEW

File: src/string/test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  });
+
+  it("generates a random without double space and right side space", function() {
+    const charset = "a      bc ";
+    const generatedString = string(integer(100), charset);
+
+    expect(generatedString.length).to.be.equal(
+      removeDoubleSpace(generatedString).length
+    );
+  });
```
Comment (ndhoule) [inline src/string/test.js]: Please add tests that exercise `fake.string`, and please also add edge cases--what if someone passes `fake.string(1|2)`?

Given that this is a random data generator, you can test this by running the function in a loop (100000 iterations should be both very fast and good enough to catch errors.)

File: src/string/index.js [hunk 2]
```diff
@@ -1,18 +1,30 @@
 // @flow
 
-import _ from 'lodash';
+import _ from "lodash";
```
[the remark this answers is not in the document]
Comment (serhalp) [reply src/string/index.js]: (We need to add prettier to this repo, I believe)

File: src/string/index.js [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
 
-import sample from '../sample';
-import integer from '../integer';
+import sample from "../sample";
+import integer from "../integer";
+
+export const removeDoubleSpace = (str: string): string => str.replace(/\s+/g, " ");
```
[the remark this answers is not in the document]
Comment (davidsonsns) [reply src/string/index.js]: I think there was a misunderstanding. The RN issue was happening because of the double spaces within the content and also the final spaces.

**Comment by ndhoule** [comment]
I'm fine with ditching this changeset and specifying a charset in MMA, or with adding this as two more functions, `fake.word`  (which would reuse `fake.string` but specify q charset with no whitespace) and `fake.sentence` (which would reuse `fake.word`).

**Comment by serhalp** [comment]
> two more functions, `fake.word`  (which would reuse `fake.string` but specify a charset with no whitespace) and `fake.sentence` (which would reuse `fake.word` and allow you to specify `times`).

👍💯

**Comment by davidsonsns** [comment]
I totally agree that it's a better way to follow. I think there was a misunderstanding because the issue is related not just to the ends/starts spaces, but to the double spaces into the string as well. I also believe that the solution proposed by @ndhoule is better. So, I'll close this PR.
